### PR TITLE
Upgrade dependency 0.49.0

### PR DESCRIPTION
**Pull request by ruthvik17**
**Description:** <Describe what has changed.>

Updated collector version to v0.49.0
**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>
All tests passed
**Documentation:** <Describe the documentation added.>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -15,10 +15,7 @@ name: C/I
 on:
   push:
     branches:
-      - main
-      - release/v*
-      - dev
-      - test/*
+      - update_canary_retry_logic
```
[the remark this answers is not in the document]
Comment (ruthvik17) [reply .github/workflows/CI.yml]: Removed the changes to CI.

File: pkg/config/config_factory.go [hunk 2]
```diff
@@ -35,5 +44,29 @@ func GetConfigProvider() service.ConfigProvider {
 		loc = []string{"env:" + envKey}
 	}
 
-	return service.MustNewDefaultConfigProvider(loc, getSetFlag())
+	providers := []config.MapProvider{filemapprovider.New(), envmapprovider.New(), yamlmapprovider.New()}
+
+	ret := make(map[string]config.MapProvider, len(providers))
```
Comment (bryan-aguilar) [inline pkg/config/config_factory.go:49]: What is `ret` supposed to symbolize? Can we choose a better variable name?

File: pkg/config/config_factory.go [hunk 3]
```diff
@@ -35,5 +44,29 @@ func GetConfigProvider() service.ConfigProvider {
...
+		Unmarshaler:   configunmarshaler.NewDefault(),
+	}
+
+	settings.MapConverters = append([]config.MapConverterFunc{overwritepropertiesmapconverter.New(getSetFlag())}, settings.MapConverters...)
+
+	config_provider, err := service.NewConfigProvider(settings)
+
+	if err != nil {
+		log.Printf("Err on creating Config Provider: %v\n", err)
+		panic(err)
```
Comment (bryan-aguilar) [inline pkg/config/config_factory.go:67]:
```suggestion
		log.Panicf("Err on creating Config Provider: %v\n", err)
```

File: pkg/config/config_factory.go [hunk 4]
```diff
@@ -35,5 +44,29 @@ func GetConfigProvider() service.ConfigProvider {
...
+
+	ret := make(map[string]config.MapProvider, len(providers))
+	for _, provider := range providers {
+		ret[provider.Scheme()] = provider
+	}
+
+	settings := service.ConfigProviderSettings{
+		Locations:     loc,
+		MapProviders:  ret,
+		MapConverters: []config.MapConverterFunc{expandmapconverter.New()},
```
Comment (bryan-aguilar) [inline pkg/config/config_factory.go:57]:
```suggestion
		MapConverters: []config.MapConverterFunc{expandmapconverter.New(),overwritepropertiesmapconverter.New(getSetFlag())},
```

File: pkg/config/config_factory.go [hunk 5]
```diff
@@ -35,5 +44,29 @@ func GetConfigProvider() service.ConfigProvider {
...
+		Unmarshaler:   configunmarshaler.NewDefault(),
+	}
+
+	settings.MapConverters = append([]config.MapConverterFunc{overwritepropertiesmapconverter.New(getSetFlag())}, settings.MapConverters...)
```
Comment (bryan-aguilar) [inline pkg/config/config_factory.go:61]:
```suggestion
```

File: pkg/config/config_factory.go (hunk 2 above)
Comment (ruthvik17) [reply pkg/config/config_factory.go]: Changed the name to mapProviders

File: pkg/config/config_factory.go (hunk 3 above)
Comment (ruthvik17) [reply pkg/config/config_factory.go]: Done

**Review by bryan-aguilar — APPROVED** [review]

File: pkg/config/config_factory.go (hunk 4 above)
Comment (ruthvik17) [reply pkg/config/config_factory.go]: Done

**Pull request merged**